### PR TITLE
Slider throws error if given 0 for stepSize or labelStepSize

### LINKED ISSUE #828: labelStepSize in slider can take 0 and crash browser
### Bug report

- __Package version(s)__: 1.11.1
- __Browser and OS versions__:  Chrome 57.0.2987.98 (64-bit) on ArchLinux x64

#### Steps to reproduce

1. Set the `labelStepSize` of a slider to 0
1. Try it in browser

#### Actual behavior

It does not perform validation on this prop.
This will crash the chrome process, showing `Aw, Snap`.

Since it simply loops by doing: 
```js
 for (let i = min, left = 0; i < max || approxEqual(i, max); i += labelStepSize, left += stepSize) {
            labels.push(<div className={`${Classes.SLIDER}-label`} key={i} style={{left}}>{this.formatLabel(i)}</div>);
        }
```

#### Expected behavior

At least `assert(labelStepSize > 0)` before looping. A better implementation would be not to show labels at all

### PR DESCRIPTION
**Pull request by giladgray**

### CODE REVIEW

File: packages/core/src/common/errors.ts [hunk 1]
```diff
@@ -50,7 +50,9 @@ export const RADIOGROUP_RADIO_CHILDREN = `${ns} <RadioGroup> only supports <Radi
 export const RADIOGROUP_CHILDREN_OPTIONS_MUTEX =
     `${ns} <RadioGroup> children and options props are mutually exclusive.`;
 
-export const RANGESLIDER_NULL_VALUE = `${ns} <RangeSlider> value prop must be an array of two non-null numbers`;
+export const SLIDER_ZERO_STEP = `${ns} <Slider> stepSize cannot be zero.`;
```
Comment (llorca) [inline packages/core/src/common/errors.ts]: Nit: no back ticks here?

Comment (cmslewis) [inline packages/core/src/common/errors.ts]: Nit: Should single-sentence errors have trailing periods or not?

File: packages/core/src/components/slider/coreSlider.tsx [hunk 2]
```diff
@@ -22,7 +23,7 @@ export interface ICoreSliderProps extends IProps {
     disabled?: boolean;
 
     /**
-     * Increment between successive labels.
+     * Increment between successive labels. Must be non-zero.
```
Comment (cmslewis) [inline packages/core/src/components/slider/coreSlider.tsx]: What happens when it's negative?

File: packages/core/test/slider/sliderTests.tsx [hunk 3]
```diff
@@ -180,6 +180,12 @@ describe("<Slider>", () => {
         });
     });
 
+    it("throws error if given 0 for stepSize or labelStepSize", () => {
```
Comment (cmslewis) [inline packages/core/test/slider/sliderTests.tsx]: nit: I would have split these into separate tests, but this is fine too.

File: packages/core/src/common/errors.ts (hunk 1 above)
Comment (giladgray) [inline packages/core/src/common/errors.ts]: _really_ don't want to make that change right now.

File: packages/core/test/slider/sliderTests.tsx (hunk 3 above)
Comment (giladgray) [inline packages/core/test/slider/sliderTests.tsx]: i mean see the one above

File: packages/core/src/common/errors.ts (hunk 1 above)
Comment (giladgray) [inline packages/core/src/common/errors.ts]: what backticks? this is a string for the console and we don't use backticks in any of the others.

File: packages/core/src/components/slider/coreSlider.tsx (hunk 2 above)
Comment (giladgray) [inline packages/core/src/components/slider/coreSlider.tsx]: negative label step crashes the browser (infinite loop again) but step size just reverse the arrow keys 😄

Comment (cmslewis) [inline packages/core/src/components/slider/coreSlider.tsx]: Should we go ahead and fix in this PR?

**Pull request merged**